src/Repository/UserRepository.php

```diff
@@ -885,6 +885,31 @@ protected function attachCriteriaToQueryBuilder(
             $qb->setParameter(':sessions', $ids);
         }
 
+        if (array_key_exists('instructedAcademicYears', $criteria)) {
+            $ids = is_array($criteria['instructedAcademicYears'])
+                ? $criteria['instructedAcademicYears'] : [$criteria['instructedAcademicYears']];
+            $qb->leftJoin('x.instructedOfferings', 'y_offering');
+            $qb->leftJoin('x.instructorIlmSessions', 'y_ilm');
+            $qb->leftJoin('x.instructorGroups', 'y_iGroup');
+            $qb->leftJoin('y_iGroup.offerings', 'y_offering2');
+            $qb->leftJoin('y_iGroup.ilmSessions', 'y_ilm2');
+            $qb->leftJoin('y_offering.session', 'y_session');
+            $qb->leftJoin('y_ilm.session', 'y_session2');
+            $qb->leftJoin('y_offering2.session', 'y_session3');
+            $qb->leftJoin('y_ilm2.session', 'y_session4');
+            $qb->leftJoin('y_session.course', 'y_course');
+            $qb->leftJoin('y_session2.course', 'y_course2');
+            $qb->leftJoin('y_session3.course', 'y_course3');
+            $qb->leftJoin('y_session4.course', 'y_course4');
+            $qb->andWhere($qb->expr()->orX(
```

Comment:
gnarly query. 

thinking out loud here - perhaps it might make sense to slap an Index on course::year to increase performance.

Reply:
Yeah, it's a copy paste job, but for sure can be better!